src/serac/physics/solid_functional.hpp

```diff
@@ -299,14 +408,28 @@ class SolidFunctional : public BasePhysics {
         [this, parameterized_material](auto x, auto displacement, auto... params) {
           auto [u, du_dX] = displacement;
 
-          auto response = parameterized_material(x, u, du_dX, serac::get<0>(params)...);
+          auto [shape, dshape_dX] = serac::solid_util::shape(Index<shape_index>{}, params...);
+
+          // Compute the displacement gradient with respect to the shape-adjusted coordinate.
+
+          // Note that the current configuration x' = X + u + p, where X is the original reference
+          // configuration, u is the displacement, and p is the shape displacement. We need the gradient with
+          // respect to the perturbed reference configuration X' = X + p for the material model. Therefore, we calculate
+          // du/dX' = du/dX * dX/dX' = du/dX * (dX'/dX)^-1 = du/dX * (I + dp/dX)^-1
+          auto du_dX_shape_mod = dot(du_dX, inv(I_ + dshape_dX));
+
+          auto response = parameterized_material(x + shape, u, du_dX_shape_mod, serac::get<0>(params)...);
 
           auto flux = 0.0 * du_dX;
 
           double geom_factor = (geom_nonlin_ == GeometricNonlinearities::On ? 1.0 : 0.0);
 
-          auto deformation_grad = du_dX + I_;
-          auto source           = response.density * u * (1.0 + geom_factor * (det(deformation_grad) - 1.0));
+          // This deformation gradient is the volumetric transform to get us back to the original
+          // reference configuration dx/dX = I + du/dX + dp/dX. If we are not including geometric
+          // nonlinearities, we ignore the du/dX factor
+          auto deformation_grad = geom_factor * du_dX + dshape_dX + I_;
+
+          auto source = response.density * u * det(deformation_grad);
 
           return serac::tuple{source, flux};
         },
```

Comment:
it looks like there are some notation inconsistencies here:
1. "shape" (in the code) vs. "p" (in the comments/LaTeX)
2. the comments use `x'` for the current configuration, but `x` in the code appears to be the reference coordinates